### PR TITLE
New plugin - Render Markup

### PR DESCRIPTION
**Pull request by nauzilus**
This plugin injects ("renders") the markup code beneath the code block so users can actually see what the markup does.

I've made some changes to download.js and components.js to include a new option `themedCSS` which indicates there are theme based CSS files for the given plugin (or language or whatever). `noCSS` and `themedCSS` are mutually exclusive, because this allows a plugin to have a "common" stylesheet (or not) and themed stylesheets (or not).

Demo: http://nauzilus-prism-render-markup.azurewebsites.net/plugins/render-markup/
Interactive demo: http://nauzilus.bitbucket.org/

I haven't really themed "funky" because I don't know how to make it look decent. Someone more design oriented is more than welcome to have a crack :+1:

### CODE REVIEW

File: download.js [hunk 1]
```diff
@@ -211,34 +247,36 @@ function getFilesSizes() {
...
-			files.forEach(function (filepath) {
-				var file = cache[filepath] = cache[filepath] || {};
-				
-				if(!file.size) {
-
-					(function(category, id) {
-					getFileSize(filepath).then(function(size) {
-						if(size) {
-							file.size = size;
-							distro.size += file.size;
```
Comment (nauzilus) [inline download.js]: As far as I could tell, `distro.size` wasn't actually used anywhere.

**Comment by Golmote** [comment]
This looks interesting. I'll try to review it properly soon.

**Comment by Golmote** [comment]
I finally took time to review this, and it looks good to me. This plugin is a really nice idea, thanks for submitting it!
@LeaVerou What do you think?

**Comment by LeaVerou** [comment]
I’m not opposed, but it looks like this PR makes way more changes than adding this plugin. Please separate your changes into individual PRs? Thanks!

**Comment by nauzilus** [comment]
Great thanks, will do! I'll close this off an submit two new PRs.

**Pull request closed without merging**